aodndata/acorn/handler.py

```diff
@@ -0,0 +1,133 @@
+
+from aodncore.pipeline import HandlerBase, PipelineFilePublishType
+from aodncore.pipeline.exceptions import InvalidFileNameError
+from aodncore.util.misc import get_pattern_subgroups_from_string
+
+FILE_TYPE_NEED_INDEX = ('radial', 'radial_quality_controlled', 'gridded_1h-avg-current-map_non-QC',
+                        'gridded_1h-avg-current-map_QC', 'gridded_1h-avg-wind-map_QC',
+                        'gridded_1h-avg-wave-site-map_QC')
+
+ACORN_FILE_PATTERN = re.compile(r"""
+                                (.*/|)IMOS_ACORN_
+                                (?P<data_parameter_code>[A-Z].*)_
+                                (?P<nc_time_cov_start>[0-9]{8}T[0-9]{6}Z)_
+                                (?P<platform_code>[A-Z]{3,4})_
+                                (?P<file_version>FV0[0-1])_
+                                (?P<product_type>.*)
+                                \.nc$
+                                """, re.VERBOSE)
+
+
+def get_type(filepath):
+    """return acorn_file_type, the file type of an ACORN file based on its filename"""
+    file_basename = os.path.basename(filepath)
+    unknown_product = False
+    if ACORN_FILE_PATTERN.match(file_basename):
+        fields = get_pattern_subgroups_from_string(file_basename, ACORN_FILE_PATTERN)
+        product_type = fields['product_type']
+        file_version = fields['file_version']
+        platform_code = fields['platform_code']
+
+        if product_type == 'radial' and file_version == 'FV00':
+            acorn_file_type = "radial"
+
+        elif product_type == 'radial' and file_version == 'FV01':
+            acorn_file_type = "radial_quality_controlled"
+
+        elif product_type == 'sea-state' and file_version == 'FV00':
+            acorn_file_type = "vector"
+
+        elif product_type == 'wavespec' and file_version == 'FV01':
+            acorn_file_type = "gridded_1h-avg-wave-spectra_QC"
+
+        elif product_type == 'windp' and file_version == 'FV01':
+            acorn_file_type = "gridded_1h-avg-wind-map_QC"
+
+        elif product_type == 'wavep' and file_version == 'FV01':
+            site_map_station = ['CBG', 'SAG', 'ROT', 'COF']
+
+            if any(s == platform_code for s in site_map_station):
+                acorn_file_type = "gridded_1h-avg-wave-site-map_QC"
+            else:
+                acorn_file_type = "gridded_1h-avg-wave-station-map_QC"
+
+        elif product_type == '1-hour-avg' and file_version == 'FV00':
+            acorn_file_type = "gridded_1h-avg-current-map_non-QC"
+
+        elif product_type == '1-hour-avg' and file_version == 'FV01':
+            acorn_file_type = "gridded_1h-avg-current-map_QC"
+
+        else:
+            unknown_product = True
+    else:
+        unknown_product = True
+
+    if unknown_product:
+        raise InvalidFileNameError("file name: \"{filename}\" Unknown product type from filename".
+                                   format(filename=os.path.basename(filepath)))
```

Comment:
You've already got the basename in a variable (file_basename), so can avoid this extra function call: